hermit-abi/src/lib.rs

```diff
@@ -1,8 +1,7 @@
 //! `hermit-abi` is small interface to call functions from the unikernel
 //! [RustyHermit](https://github.com/hermitcore/libhermit-rs).
 
-#![cfg_attr(feature = "rustc-dep-of-std", no_std)]
```

Comment:
I'm fairly sure that this line is needed, since the hermit `std` library (in the rust compiler) depends on the hermit-abi crate.

Reply:
This just removes the `cfg_attr` so that this crate is always `no_std`, so things should still work correctly with it being a dep of libstd.

Reply:
AFAIK as of now, it is still necessary to have `rustc-dep-of-std` in dependencies of `std` library crates.
There is a plan to remove it (https://github.com/rust-lang/wg-cargo-std-aware/issues/51), but the issue is still open.

Reply:
That correct, and that feature isn't removed by this PR

Reply:
A sorry, I got confused there for a moment. Thanks for the clarification.

Reply:
For example, `libc` (which is also a dependency of `std`) is just unconditionally `no_std`, see: https://github.com/rust-lang/libc/blob/715b50e1555a339d4e474ac4f6ae5a944bc1b180/src/lib.rs#L28

Reply:
The current network support depends on `std`.  `libstd` doesn't select this feature. But we should make sure that this is not enabled if `std` is not enabled.

Reply:
It looks like that support is in a different crate, this crate (`hermit-abi`) is just a facade crate that doesn't depend on any std code or types.

Reply:
I will fix with another PR.

Reply:
> It looks like that support is in a different crate, this crate (`hermit-abi`) is just a facade crate that doesn't depend on any std code or types.

Sorry, my mistake. I mixed up `hermit-sys` and `hermit-abi`.  :-)